src/facet/inspection/_explainer.py

```diff
@@ -309,7 +546,7 @@ def uses_background_dataset(self) -> bool:
         """[see superclass]"""
         return True
 
-    def make_explainer(self, model: LearnerDF, data: pd.DataFrame) -> Explainer:
+    def make_explainer(self, model: LearnerDF, data: pd.DataFrame) -> BaseExplainer:
         """[see superclass]"""
 
         self._validate_background_dataset(data=data)
```

Comment:
I'm always in favour of additional checks that allow for a "fail fast" approach. I would raise a `ValueError` if the model is not fitted, instead of using an assertion: assertions should be reserved to testing conditions we consider guaranteed to be true; hence, a failed assertion is indicative to a bug in the library but not to wrong use by the end user.

Reply:
Yes, I agree! `self._ensure_fitted()` will be better than `assert` as already used in the codebase.

Reply:
Given this will not be a call to `self`, probably rather
```python
if not model.is_fitted():
  raise ValueError("arg model must be fitted")
```